### ISSUE TITLE
Censor-Resistant Web Hosting Services

### ISSUE THREAD

**Issue by njmurarka** [bluzelle/Gitcoin10Hack#1]
### Prize Title

Censor-Resistant Web Hosting Services (being deprecated and replaced with a new bounty)

### Prize Bounty

$5,500 USD worth of ERC20 BLZ, at time of payout

### Challenge Description

Bluzelle provides decentralized data services, powered by the Cosmos blockchain. Our services include a key-value-store (CRUD), oracle, and NFT. We are also building toward providing support to EVM (Ethereum Virtual Machine) and Polkadot support for our services. Our bounties reflect our aggressive approach to consistently improve our ecosystem and value proposition.

While the Hackathon has a specific start and end date, we are ok with work continuing after the hackathon, for the chosen winners to finish their projects to our standards. 

A service that sits on top of BLZ.JS and provides web-specific primitives to push and pull updates to a website, somewhat similar to how Git works. The user of this service would be expected to create their own BLZ account, fund it with BLZ tokens, and configure their DNS as needed (ie: point a CNAME record to the special endpoint where these services live). They can then use this service to push and pull content, very much like Git.

In doing this, the content is now stored immutably on the Bluzelle DB, and is hosted on the fly and transparently to customers of that website. The service is in fact allowing for authoring and is also hosting requests for the website. 

End result is you can host an entire website on our infrastructure, free from the risk of takedown by any 3rd parties (contingent on a proper decentralized design). This is a particularly powerful API upon which free-speech and other related platforms can be built. While we cannot vouch for something like “Parler”, an app such as this one could be built on our platform and not face the risk of takedown by anybody.

### Resources

Our Discord: 

https://discord.gg/yqBmzPxRZK

Website:

https://www.bluzelle.com

Our JS library:

https://www.npmjs.com/package/@bluzelle/sdk-js

To install our libraries:

install @bluzelle/sdk-js with “yarn” or “npm”

### Submission Requirements

The submission should include sufficiently QA’d documentation on how to deploy the service/product, and how to use the submission as per the requirements of the bounty. 

These should include documentation on the commands to be used to interact with the submission, and how the submission is configured to work properly with BluzelleDB, etc.

A video demo should be included. It would nice to have a voice-over in English where we can fully understand the submission, but this is not a strict requirement. A computer-generated voice over is ok too, if you prefer. 

The demo should also walk through the code and explain all the items that are being provided. The demo should walk through the process of deploying the submission, and how to use it, etc.

It is expected that the documentation is accurate. We will follow your documentation, to properly evaluate the submission. If it is incorrect, we may be unable to fairly evaluate your submission.

Including tests with your submission will greatly improve your chances of winning. We like to run tests that are highly verbose and explicit in terms of what they are doing, so we can gain confidence in the correctness of what you have submitted. If you provide tests and expect us to run them, like everything else, document it well, and ensure that the tests can be run by us -- give us the steps to setup and run the tests.

If the documentation is incomplete or incorrect, there is a possibility that we may not be able to fairly assess the submission, as we will walk through the documentation to validate the project. Due to practical limitations on time and resources, once a project is submitted, we are not able to provide much assistance in correcting a project’s that may not be properly working, nor to inquire to get proper steps, if the documentation that comes with a submission is insufficient. 

Your project will be judged based on what you submit. Please submit something that is complete, well thought out, and tested, from a documentation and product features and code quality standpoint. We will do our best to evaluate, but obviously, the easier you make our life, the better the chances are that you win. 

WE LOVE VERBOSITY AND DOCUMENTATION. There is no such thing as too much information. Explain what you have built, and please ensure it will run CORRECTLY, when we follow your directions literally. Just doing this alone will vastly improve your chances of victory.

### Judging Criteria

Our goal is to, as part of the evaluation process, fully setup, and use your submission, successfully, and without any major hiccups.

Based on the ease of doing this and the quality of your documentation, product, code, and features, we will choose the winner. 

There is no preference to ordering of submissions -- just be sure to submit them on time. Once submitted, we will evaluate and there will not be alot of opportunity for back and forth. Please ensure your submitted documentation and code is complete, enabling us to properly judge it based on its merit.

We will choose the best based on quality. Documentation and properly written code is a large part of the criteria. A project that we cannot deploy ourselves is difficult to give a prize to. We will do our best to contact you, if there is an issue. Practically, we probably won’t have much time to contact you, after submission, to get clarification or to ask you to fix a bug. It ideally should work when we judge it.

Note: While the descriptions given for bounties are quite explicit and even tend to suggest how an actual solution to each problem can be built, you as a developer have the option to architect the solution your own way. We have provided guidance on a solution we see as reasonable, but we are open to considering other solutions. Obviously, we will choose the best overall submission, based on various factors including the elegance of the solution.

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__This issue now has a funding of 26829.0 BLZ (3723.87 USD @ $0.14/BLZ)  attached to it.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/bluzelle/Gitcoin10Hack/1/100025952).
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/bluzelle/Gitcoin10Hack/1/100025952).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://discord.gg/gitcoin/'>Gitcoin's Discord</a>
* $2,231,478.43 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Cancelled** 

<hr>

__Work has been started__.


These users each claimed they can complete the work by 2 days, 8 hours from now.
Please review their action plans below:


**1) [osamaelhariri](https://gitcoin.co/osamaelhariri) has started work.**

Create a web page server that sends clients files that are stored in Bluzelle. There will also be a tool that allows developers to upload and update their web page.

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/bluzelle/Gitcoin10Hack/1/100025952).

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__The funding of 5500.0 USDC  (5500.0 USD @ $1.0/USDC) attached to this issue has been **cancelled** by the bounty submitter__

 
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://discord.gg/gitcoin/'>Gitcoin's Discord</a>
* $2,067,388.15 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by njmurarka** [comment]
Replaced with:

https://github.com/bluzelle/Gitcoin10Hack/issues/9
